apps/src/templates/instructions/ResourceLink.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { PropTypes } from 'react';
+import Radium from 'radium';
+import color from '@cdo/apps/util/color';
+import BaseDialog from '../BaseDialog';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+
+const styles = {
+  textLink: {
+    display: 'inline-block',
+    margin: 8,
+    fontWeight: 'bold',
+    fontSize: 16,
+    lineHeight: '25px',
+    cursor: 'pointer',
+    maxWidth: '90%'
+  },
+  mapThumbnail: {
+    backgroundColor: color.teal,
+  },
+  commonThumbnail: {
+    borderRadius: 5,
+    paddingLeft: 26,
+    paddingRight: 26,
+    paddingTop: 16,
+    paddingBottom: 9,
+  },
+  commonIcon: {
+    fontSize: 22,
+  },
+  mapIcon: {
+    color: color.white
+  },
+  resourceIcon: {
+    color: color.teal
+  },
+  resourceStyle: {
+    margin: 8
+  },
+  linkFrame: {
+    width: '98%',
+    height: '94%'
+  }
+};
+
+class ResourceLink extends React.Component {
+  static propTypes = {
+    highlight: PropTypes.bool,
+    icon: PropTypes.string.isRequired,
+    text: PropTypes.string.isRequired,
+    reference: PropTypes.string.isRequired,
+  };
+
+  state = {
+    dialogSelected: false
+  };
+
+  selectResource = () => {
+    this.setState({dialogSelected: true});
+  };
+
+  closeResource = () => {
+    this.setState({dialogSelected: false});
+  };
+
+  render() {
+    const { icon, reference, text, highlight} = this.props;
```

Comment:
Unbalanced whitespace.

We don't have a preference for `{ spaces }` or `{noSpaces}` but would like to avoid `{ unbalanced}`. Unfortunately, [ESLint can't enforce that](https://eslint.org/docs/rules/object-curly-spacing) :cry:.